kolibri/plugins/video_mp4_render/assets/src/vue/index.vue

```diff
@@ -1,15 +1,223 @@
 <template>
 
-  <video autoplay=true src='http://vid297.photobucket.com/albums/mm238/daystar170/RickRoll.mp4' controls=true></video>
+  <div>
+    <div class="videowrapper" v-el:videowrapper>
+      <video v-el:video class="video-js vjs-default-skin" >
+        <source :src="videoUrl" type='video/mp4'>
+        <track kind="captions" :src="captionsUrl" :srclang="captionsLang" :label="captionsLabel">
+      </video>
+    </div>
+  </div>
 
 </template>
 
 
 <script>
 
-  module.exports = {};
+  const videojs = require('video.js');
+  require('./replayPlayForward');
+  let videoPlayer = null;
+  module.exports = {
+
+    ready() {
+      videoPlayer = videojs(this.$els.video, {
+        controls: true,
+        autoplay: false,
+        fluid: true,
+        preload: 'auto',
+        poster: this.posterUrl,
+        playbackRates: [0.25, 0.5, 1.0, 1.25, 1.5, 2.0],
+        textTrackDisplay: true,
+        ReplayButton: true,
+        ForwardButton: true,
+        TogglePlayButton: true,
+        bigPlayButton: false,
+        controlBar: {
+          children: [
+            {
+              name: 'currentTimeDisplay',
+            },
+            {
+              name: 'timeDivider',
+            },
+            {
+              name: 'progressControl',
+            },
+            {
+              name: 'durationDisplay',
+            },
+            {
+              name: 'remainingTimeDisplay',
+            },
+            {
+              name: 'muteToggle',
+            },
+            {
+              name: 'VolumeBar',
+            },
+            {
+              name: 'playbackRateMenuButton',
+            },
+            {
+              name: 'captionsButton',
+            },
+            {
+              name: 'fullscreenToggle',
+            },
+          ],
```

Comment:
this can be written a bit more compactly as

```javascript
          children: [
            { name: 'currentTimeDisplay' },
            { name: 'timeDivider' },
            { name: 'progressControl' },
            { name: 'durationDisplay' },
            { name: 'remainingTimeDisplay' },
            { name: 'muteToggle' },
            { name: 'VolumeBar' },
            { name: 'playbackRateMenuButton' },
            { name: 'captionsButton' },
            { name: 'fullscreenToggle' },
          ],
```